extensions/resume.md

```diff
@@ -0,0 +1,152 @@
+
+Other clients that have the reconnecting user `MONITOR`'d MUST be sent one `RPL_MONOFFLINE` numeric and one `RPL_MONONLINE` numeric indicating that the user reconnected.
+
+Upon resuming the connection, servers close the old connection.
+
+### TLS
+
+Clients may initially connect with TLS, only to later attempt to `RESUME` from a connection without TLS. In this situation, servers SHOULD deny the attempt with a message such as _"Cannot resume connection, client is no longer using TLS"_.
+
+This is intended to ensure that TLS-only channel modes are not bypassed, and that clients prefer continuing to use TLS.
+
+
+## Examples
+
+A client with the nick `dan` reconnecting. The old connection used the username `~old` and the host `192.168.0.5`, and the new connection uses the username `~d` and the host `127.0.0.1`:
+
+    C1 - C: PING 12345678
+    C1 - S: :irc.example.com PONG 12345678
+    C1 - C: PING 87654321
+    ... C1 - Server does not send a response ...
+    ... C1 disconnects and reconnects as C2 ...
+
+    C2 - C: CAP LS
+    C2 - C: NICK dan-
+    C2 - C: USER d * 0 :An example user!
+    C2 - S: :irc.example.com CAP * LS :sasl draft/resume
+    C2 - C: CAP REQ :sasl draft/resume
+    C2 - S: :irc.example.com CAP dan- ACK :sasl draft/resume
+    C2 - C: RESUME dan 2017-04-13T15:12:51.620Z
+    C2 - C: AUTHENTICATE PLAIN
+    C2 - S: :irc.example.com AUTHENTICATE +
+    C2 - C: AUTHENTICATE YnVubnkAYnVubnkAYnVubnk=
+    C2 - S: :irc.example.com 900 dan- * bunny :You are now logged in as bunny
+    C2 - S: :irc.example.com 903 dan- :SASL authentication successful
+    C2 - S: :irc.example.com RESUMED dan ~d 127.0.0.1 2017-04-13T15:12:51.620Z
+    ... C1's connection is closed and C1's attributes are applied to C2 ...
+    C2 - S: :irc.example.com 001 dan :Welcome to the Internet Relay Network dan
+    ... C2 receives regular registration burst ...
+    C2 - S: :irc.example.com 376 dan :End of MOTD command
+    C2 - S: :dan!~d@127.0.0.1 JOIN #test
+    C2 - S: :irc.example.com 332 dan #test :Example topic
+    C2 - S: :irc.example.com 333 dan #test george 1442060874
+    C2 - S: :irc.example.com 353 dan @ #test :@dan @george +violet roger
+    C2 - S: :irc.example.com MODE #test +o dan
+
+Here is this reconnection seen by `george`, a client that does not have the `draft/resume` capability enabled:
+
+    C: :dan!~old@192.168.0.5 QUIT :Client reconnected
+    S: :dan!~d@127.0.0.1 JOIN #test
+    S: :irc.example.com MODE #test +o dan
+
+And here is this reconnection seen by `violet`, a client that has the `draft/resume` capability:
+
+    C: :dan!~old@192.168.0.5 RESUMED dan ~d 127.0.0.1 2017-04-13T15:12:51.620Z
```

Comment:
Isn't this RESUMED essentially the same as CHGHOST with an additional timestamp? Not sure how that timestamp could be useful for other clients.